lib/factbase/sync/sync_factbase.rb

```diff
@@ -57,10 +57,11 @@ def txn
 
   private
 
-  def try_lock
-    locked = @mutex.try_lock
-    r = yield
-    @mutex.unlock if locked
-    r
+  def try_lock(&)
+    if @mutex.owned?
```

Comment:
@Yegorov what if `.owned?` returns `true` and then immediately after that the locks gets released and acquired by another thread? Here, we will continue and do `yield`, while the lock is not ours anymore.